### PR TITLE
fix(provider): update checkpoints only for known stages

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Update stage progress only for known stage ids as opposed to all entries in the table.

_Labeled as bug because of unexpected behavior._

### CODE REVIEW

**Comment by gakonst** [comment]
what was the bug / when did it happen?

**Comment by rkrasiuk** [comment]
@gakonst this happened as I was attempting to update log indexing. Now, I basically have to resync that stage

**Comment by onbjerg** [comment]
Ah, right, if a new stage is added and the tree is run, it would update the new stage to the latest block number (even though it hasn't synced) - I guess that's the bug, right? Good catch

**Comment by rkrasiuk** [comment]
@onbjerg at least the stage wouldn't be inconsistent. i opened https://github.com/paradigmxyz/reth/pull/3623 in order to understand better why this behavior occurred

File: crates/storage/provider/src/providers/database/provider.rs [hunk 1]
```diff
@@ -1214,14 +1214,15 @@ impl<'this, TX: DbTxMut<'this>> StageCheckpointWriter for DatabaseProvider<'this
     ) -> Result<()> {
         // iterate over all existing stages in the table and update its progress.
```
Comment (mattsse) [inline crates/storage/provider/src/providers/database/provider.rs:1215]: is this doc this accurate?

Comment (onbjerg) [reply crates/storage/provider/src/providers/database/provider.rs:1215]: yes

**Review by onbjerg — APPROVED** [review]

**Pull request merged**